src/wp-includes/cron.php

```diff
@@ -417,11 +417,15 @@ function wp_reschedule_event( $timestamp, $recurrence, $hook, $args = array(), $
 	}
 
 	// Now we assume something is wrong and fail to schedule.
-	if ( 0 === $interval ) {
+	if ( ! is_int( $interval ) || $interval <= 0 ) {
```

Comment:
```suggestion
	if ( ! is_numeric( $interval ) || $interval <= 0 ) {
```

Matches the behaivour of the existing check in `wp_schedule_event()`